.evergreen/config.yml

```diff
@@ -444,8 +444,29 @@ functions:
         silent: true
         working_dir: "src"
         script: |
-          # DO NOT ECHO WITH XTRACE (which PREPARE_SHELL does)
-          PYTHON_BINARY=${PYTHON_BINARY} ATLAS_REPL='${atlas_repl}' ATLAS_SHRD='${atlas_shrd}' ATLAS_FREE='${atlas_free}' ATLAS_TLS11='${atlas_tls11}' ATLAS_TLS12='${atlas_tls12}'  sh ${PROJECT_DIRECTORY}/.evergreen/run-atlas-tests.sh
+          cat <<EOT > prepare_atlas_connectivity.sh
+          export ATLAS_FREE='${atlas_free}'
```

Comment:
I think we can further optimize some of these invocations by switching to using `subprocess.exec` instead of `shell.exec` and then setting either the`include_expansions_in_env` option or the `add_expansions_to_env` option to automatically set the appropriate environment variables. I suppose we don't need to do it right now, but being able to eventually eliminate the use of `$PREPARE_SHELL` would be a huge improvement IMO.

Reply:
Just to clarify, I am just suggesting this as a future optimization/refactor. Not saying that we need to do it in this PR. Would love to hear your thoughts on it @ShaneHarvey

Reply:
Cool idea! Can you open a ticket with this comment?

Reply:
https://jira.mongodb.org/browse/PYTHON-2481